src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -65,6 +65,15 @@ class DockerImageOptionValueError(ValueError):
 
 @dataclass(frozen=True)
 class DockerFieldSet(PackageFieldSet, RunFieldSet):
+    """Builds a Docker image from a Dockerfile.
+
+    The target fields (e.g. the `docker_image` fields) and the `[docker]` configuration scope
+    provides the required details for building the image.
+
+    See `./pants help docker docker_image` for avilable options, fields and their respective
+    documentations.
+    """
```

Comment:
I'm not sure how valuable this is to document, tbh. There doesn't seem to be anything note worthy about this class - it's a FieldSet like any others.

> See `./pants help docker docker_image` for avilable options, fields and their respective
      documentations.
      
These docs are for plugin authors, so if we wanted something like this, it would be better to point to the code.

--

The reason I'm nervous about this is it's really hard to keep docs up-to-date. This is why our style guide generally prefers self-documenting code over comments: https://www.pantsbuild.org/docs/style-guide#when-to-comment

Reply:
It was added to test out how to document the types for the CLI help output. We can remove it or at least brief it down if you prefer.

Reply:
Okay. I think with this one, probably remove the docstring

Reply:
OK, fwiw, this is what it looked like... :) (from #14227 ). 
![image](https://user-images.githubusercontent.com/72965/170743928-298f8bc0-a7d1-4e5b-9b2b-5fd85f9a4c53.png)